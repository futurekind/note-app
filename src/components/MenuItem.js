import React, {PropTypes} from 'react';
import styled from 'styled-components';

const View = styled.div``;

const Title = styled.div``;

const Subtitle = styled.div``;

const Meta = styled.div``;

const Flag = styled.span``

const MenuItem = ({
    onClick,
    title,
    subtitle,
    flagColor,
    active,
    index
}) => {
    return (
        <View onClick={ () => onClick(index) } active={ active }>
            <Title>{ title }</Title>
            <Meta>
                <Subtitle>{ subtitle }</Subtitle>
                <Flag bgColor={ flagColor } />
            </Meta>
        </View>
    )
}

MenuItem.defaultProps = {
    onClick: () => {}
}

MenuItem.propTypes = {
    index: PropTypes.number.isRequired,
    active: PropTypes.bool,
    title: PropTypes.string,
    subtitle: PropTypes.string,
    flagColor: PropTypes.string,
    onClick: PropTypes.func,
}

export default MenuItem