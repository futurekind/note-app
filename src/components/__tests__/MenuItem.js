import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import MenuItem from '../MenuItem';

describe('MenuItem', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<MenuItem index={100} />)
    })

    it('renders', () => {
        expect(toJSON(wrapper)).toMatchSnapshot();
    })

    it('renders title', () => {
        wrapper.setProps({
            title: 'Some Title'
        })

        const title = wrapper.children().at(0)

        expect(toJSON(title)).toMatchSnapshot()
    })

    it('renders subtitle', () => {
        wrapper.setProps({
            subtitle: 'Some Subtitle'
        })

        const title = wrapper.children().at(1)

        expect(toJSON(title)).toMatchSnapshot()
    })

    it('renders flag color', () => {
        wrapper.setProps({
            flagColor: 'red'
        })

        const flag = wrapper
                        .children().at(1)
                        .children().at(1)

        expect(toJSON(flag)).toMatchSnapshot()
    })

    it('delegates click to onClick prop', () => {
        const spy = jest.fn();

        wrapper.setProps({
            onClick: spy
        })

        wrapper.simulate('click');

        expect(spy).toHaveBeenLastCalledWith(100)
    })

    it('renders active prop', () => {
        wrapper.setProps({
            active: true
        })

        expect(toJSON(wrapper)).toMatchSnapshot()
    })
})