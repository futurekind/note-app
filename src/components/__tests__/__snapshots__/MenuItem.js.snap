exports[`MenuItem renders 1`] = `
<styled.div
  onClick={[Function]}>
  <styled.div />
  <styled.div>
    <styled.div />
    <styled.span />
  </styled.div>
  <Divider />
</styled.div>
`;

exports[`MenuItem renders active prop 1`] = `
<styled.div
  active={true}
  onClick={[Function]}>
  <styled.div />
  <styled.div>
    <styled.div />
    <styled.span />
  </styled.div>
  <Divider />
</styled.div>
`;

exports[`MenuItem renders flag color 1`] = `
<styled.span
  bgColor="red" />
`;

exports[`MenuItem renders subtitle 1`] = `
<styled.div>
  <styled.div>
    Some Subtitle
  </styled.div>
  <styled.span />
</styled.div>
`;

exports[`MenuItem renders title 1`] = `
<styled.div>
  Some Title
</styled.div>
`;
